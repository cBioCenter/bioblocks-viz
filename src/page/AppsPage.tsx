import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header, List } from 'semantic-ui-react';

import { IVizSummaryData, VizData } from '~chell-viz~/data';

export interface IAppsPageProps extends Partial<RouteComponentProps> {}

export class AppsPage extends React.Component<IAppsPageProps, any> {
  constructor(props: IAppsPageProps) {
    super(props);
  }

  public render() {
    return (
      <List divided={true}>
        <List.Item>{this.renderAppItem(VizData.spring)}</List.Item>
        <List.Item>{this.renderAppItem(VizData.tfjsTsne)}</List.Item>
        <List.Item>{this.renderAppItem(VizData.anatomogram)}</List.Item>
      </List>
    );
  }

  protected renderAppItem(viz: IVizSummaryData) {
    return (
      <Grid centered={true} columns={3} padded={true} relaxed={true}>
        <Grid.Column>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            alt={`icon for ${viz.name}`}
            style={{ height: '100px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h2'}>{viz.name}</Header>
          <List>
            <List.Item>{`${viz.listAsOriginal ? 'original: ' : ''}${viz.authors.join(', ')}`}</List.Item>
            <List.Item>{viz.summary}</List.Item>
            <List.Item>{`relevant data: ${viz.relevantData}`}</List.Item>
          </List>
        </Grid.Column>
        <Grid.Column stretched={true}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/overview', search: `?name=${viz.name.toLocaleLowerCase()}` }}>details</Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={'/beta'}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid>
    );
  }
}
