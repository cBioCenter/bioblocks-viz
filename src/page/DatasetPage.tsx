import { fromJS, List } from 'immutable';
import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Grid, Message } from 'semantic-ui-react';

import { AnatomogramContainer, SpringContainer, TensorTContainer } from '~chell-viz~/container';
import { VizData } from '~chell-viz~/data';

export interface IDatasetPageProps extends Partial<RouteComponentProps> {}

export interface IDatasetPageState {
  apps: List<string>;
  datasetLocation: string;
}

export class DatasetPage extends React.Component<IDatasetPageProps, IDatasetPageState> {
  constructor(props: IDatasetPageProps) {
    super(props);
    this.state = {
      apps: List<string>(),
      datasetLocation: '',
    };
  }

  public componentDidMount() {
    console.log(this.props);
    if (this.props.location) {
      this.setupSearchParameters(this.props.location.search);
    }
  }

  public componentDidUpdate(prevProps: IDatasetPageProps) {
    if (this.props.location && this.props.location !== prevProps.location) {
      this.setupSearchParameters(this.props.location.search);
    }
  }

  public render() {
    const { apps, datasetLocation } = this.state;

    return (
      <div style={{ padding: '20px' }}>
        <Grid centered={true} stackable={true} stretched={false} padded={true} columns={2}>
          {datasetLocation.length >= 1 &&
            apps.map((app, index) => (
              <Grid.Column key={`dataset-app-${index}`} style={{ width: 'auto' }}>
                {this.renderApp(app, datasetLocation)}
              </Grid.Column>
            ))}
        </Grid>
      </div>
    );
  }

  protected setupSearchParameters(query: string) {
    const params = new URLSearchParams(query);
    // tslint:disable-next-line:no-backbone-get-set-outside-model
    const datasetLocation = params.get('name');
    const apps = fromJS(params.getAll('app')) as List<string>;

    this.setState({
      apps,
      datasetLocation: datasetLocation ? datasetLocation : '',
    });
  }

  protected renderApp(app: string | undefined, datasetLocation: string) {
    switch (app) {
      case VizData.spring.name.toLocaleLowerCase():
        return <SpringContainer datasetLocation={datasetLocation} />;
      case VizData.tfjsTsne.name.toLocaleLowerCase():
        return <TensorTContainer datasetLocation={datasetLocation} />;
      case VizData.anatomogram.name.toLocaleLowerCase():
        return <AnatomogramContainer />;
      default:
        return <Message error={true}>{`Currently unsupported app '${app}'`}</Message>;
    }
  }
}
