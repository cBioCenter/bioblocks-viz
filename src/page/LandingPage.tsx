import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Container, Divider, Grid, Header, Segment } from 'semantic-ui-react';

import { IDatasetInfo, IStory, IVizSummaryData, Stories, userDatasets, VizData } from '~chell-viz~/data';

export interface ILandingPageProps extends Partial<RouteComponentProps> {}

export class LandingPage extends React.Component<ILandingPageProps, any> {
  constructor(props: ILandingPageProps) {
    super(props);
  }

  public render() {
    return (
      <Container fluid={false}>
        <Segment basic={true} padded={'very'}>
          <Grid centered={true} padded={true} relaxed={true}>
            {this.renderFeaturedStories()}
            {this.renderFeaturedVisualizations()}
            {this.renderFeaturedDatasets()}
          </Grid>
        </Segment>
      </Container>
    );
  }

  protected renderSingleVisualization(viz: IVizSummaryData) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            style={{ height: '90px', width: '90px' }}
            alt={`viz ${viz.name} icon`}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{viz.name}</Header>
          {viz.summary}
          <p>
            <span style={{ fontWeight: 'bold' }}>{viz.listAsOriginal ? 'Original authors' : 'Authors'}: </span>
            {viz.authors.length === 2 ? `${viz.authors[0]} and ${viz.authors[1]}` : viz.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column width={2}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/visualizations/', search: `?name=${viz.name.toLocaleLowerCase()}` }}>
                details
              </Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/dataset', search: `?viz=${viz.name.toLocaleLowerCase()}` }}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderFeaturedDatasets() {
    return (
      <>
        <Grid.Row centered={true}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Datasets</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={2} divided={true}>
          {this.renderHCADatasets()}
          {this.renderUserSharedDatasets()}
        </Grid.Row>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'datasets'}>
              more datasets...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedStories() {
    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Stories</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {this.renderSingleFeaturedStory(Stories[0])}
        {this.renderSingleFeaturedStory(Stories[1])}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'stories'}>
              more stories...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedVisualizations() {
    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Visualizations</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {this.renderSingleVisualization(VizData.spring)}
        {this.renderSingleVisualization(VizData.tfjsTsne)}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'visualizations'}>
              more visualizations...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderSingleFeaturedStory(story: IStory) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img src={story.icon} style={{ height: '90px', width: '90px' }} alt={`story ${story.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{story.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {story.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {story.authors.length === 2 ? `${story.authors[0]} and ${story.authors[1]}` : story.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column width={2}>
          <Button basic={true}>
            <Link to={story.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderHCADatasets() {
    return (
      <Grid.Column width={3}>
        <Grid.Row>
          <img
            src={'assets/icons/hca-logo.png'}
            style={{ width: '150px', height: '250px' }}
            alt={'human cell atlas logo'}
          />
          <Grid.Row>
            <a href={'https://preview.data.humancellatlas.org/'}>Search HCA datasets ...</a>
          </Grid.Row>
        </Grid.Row>
      </Grid.Column>
    );
  }

  protected renderSingleUserSharedDatasets(dataset: IDatasetInfo) {
    return (
      <Grid.Row columns={2}>
        <Grid.Column floated={'left'}>
          <Header>{dataset.name}</Header>
          <p>
            {dataset.summary}
            <br />
            <span style={{ fontWeight: 'bold' }}>Authors: </span>
            {dataset.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column floated={'right'}>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.detail}>details</Link>
          </Button>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.analysis}>analyze</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderUserSharedDatasets() {
    return (
      <Grid.Column stretched={true} textAlign={'left'} width={8}>
        <Grid.Row textAlign={'center'}>
          <Header as={'h2'} textAlign={'center'}>
            User shared
          </Header>
        </Grid.Row>
        <br />
        <Grid.Row>{this.renderSingleUserSharedDatasets(userDatasets[0])}</Grid.Row>
        <Grid.Row>{this.renderSingleUserSharedDatasets(userDatasets[1])}</Grid.Row>
      </Grid.Column>
    );
  }
}
