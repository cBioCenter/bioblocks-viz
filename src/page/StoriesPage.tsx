import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header } from 'semantic-ui-react';

import { IStory, Stories } from '~chell-viz~/data';

export interface IStoriesPageProps extends Partial<RouteComponentProps> {}

export class StoriesPage extends React.Component<IStoriesPageProps, any> {
  constructor(props: IStoriesPageProps) {
    super(props);
  }

  public render() {
    return (
      <Grid centered={true} divided={'vertically'} padded={true} relaxed={false}>
        {Stories.map((story, index) => (
          <React.Fragment key={`story-${index}`}>{this.renderSingleFeaturedStory(story)}</React.Fragment>
        ))}
      </Grid>
    );
  }

  protected renderSingleFeaturedStory(story: IStory) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column>
          <img src={story.icon} style={{ height: '90px', width: '90px' }} alt={`story ${story.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header>{story.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {story.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {story.authors.length === 2 ? `${story.authors[0]} and ${story.authors[1]}` : story.authors.join(', ')}
          </p>
          <br />
        </Grid.Column>
        <Grid.Column>
          <Button basic={true}>
            <Link to={story.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }
}
