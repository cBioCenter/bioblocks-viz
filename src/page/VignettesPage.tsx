import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header } from 'semantic-ui-react';

import { IVignette, Vignettes } from '~chell-viz~/data';

export interface IVignettesPageProps extends Partial<RouteComponentProps> {}

export class VignettesPage extends React.Component<IVignettesPageProps, any> {
  constructor(props: IVignettesPageProps) {
    super(props);
  }

  public render() {
    return (
      <Grid centered={true} padded={true} relaxed={false}>
        {Vignettes.map((vignette, index) => (
          <React.Fragment key={`vignette-${index}`}>{this.renderSingleFeaturedVignette(vignette)}</React.Fragment>
        ))}
      </Grid>
    );
  }

  protected renderSingleFeaturedVignette(vignette: IVignette) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column>
          <img src={vignette.icon} style={{ height: '90px', width: '90px' }} alt={`vignette ${vignette.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header>{vignette.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {vignette.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {vignette.authors.length === 2
              ? `${vignette.authors[0]} and ${vignette.authors[1]}`
              : vignette.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column>
          <Button basic={true}>
            <Link to={vignette.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }
}
