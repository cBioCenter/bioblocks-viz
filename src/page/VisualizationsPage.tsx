import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header, List } from 'semantic-ui-react';

import { IVizSummaryData, VizData } from '~bioblocks-viz~/data';

export interface IVisualizationsPageProps extends Partial<RouteComponentProps> {}

export class VisualizationsPage extends React.Component<IVisualizationsPageProps, any> {
  constructor(props: IVisualizationsPageProps) {
    super(props);
  }

  public render() {
    return (
      <List divided={true}>
        <List.Item>{this.renderVisualizationItem(VizData.spring)}</List.Item>
        <List.Item>{this.renderVisualizationItem(VizData.tfjsTsne)}</List.Item>
        <List.Item>{this.renderVisualizationItem(VizData.anatomogram)}</List.Item>
      </List>
    );
  }

  protected renderVisualizationItem(viz: IVizSummaryData) {
    return (
      <Grid centered={true} columns={3} padded={true} relaxed={true}>
        <Grid.Column>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            alt={`icon for ${viz.name}`}
            style={{ height: '100px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h2'}>{viz.name}</Header>
          <List>
            <List.Item>{`${viz.listAsOriginal ? 'original: ' : ''}${viz.authors.join(', ')}`}</List.Item>
            <List.Item>{viz.summary}</List.Item>
            <List.Item>{`relevant data: ${viz.relevantData}`}</List.Item>
          </List>
        </Grid.Column>
        <Grid.Column stretched={true}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/visualizations/', search: `?name=${viz.name.toLocaleLowerCase()}` }}>
                details
              </Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/dataset', search: `?viz=${viz.name.toLocaleLowerCase()}` }}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid>
    );
  }
}
