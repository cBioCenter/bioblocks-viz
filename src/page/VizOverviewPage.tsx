import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Accordion, Button, Container, Divider, Grid, Header, Icon, List, Message } from 'semantic-ui-react';

import { Link } from 'react-router-dom';
import { IVizOverviewData, VizData } from '~chell-viz~/data';

export interface IVizOverviewPageProps extends Partial<RouteComponentProps> {}

export interface IVizOverviewPageState {
  currentViz: IVizOverviewData | null;
}

export class VizOverviewPage extends React.Component<IVizOverviewPageProps, IVizOverviewPageState> {
  constructor(props: IVizOverviewPageProps) {
    super(props);
    this.state = {
      currentViz: null,
    };
  }

  public componentDidMount() {
    this.setupCurrentViz();
  }

  public componentDidUpdate(prevProps: IVizOverviewPageProps) {
    if (this.props.location !== prevProps.location) {
      this.setupCurrentViz();
    }
  }

  public render() {
    return (
      this.state.currentViz && (
        <Container>
          {this.renderOverview(this.state.currentViz)}
          {this.renderExamples()}
        </Container>
      )
    );
  }

  protected renderOverview(viz: IVizOverviewData) {
    return (
      <Grid centered={true} columns={2}>
        <Grid.Column width={2}>
          <img
            alt={`icon for ${viz.name}`}
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-icon.png`}
            style={{ height: '200px', padding: '20px', width: '128px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h1'}>
            {viz.name}
            <Header.Subheader>{viz.authors.join(', ')}</Header.Subheader>
          </Header>
          <>
            <p>{viz.detailedSummary}</p>
            <List>
              <List.Item>applicable data: {viz.relevantData}</List.Item>
              <List.Item>compatible with: {viz.compatibility.join(', ')}</List.Item>
              <List.Item>
                citation(s):{' '}
                {viz.citations.map((citation, index) => (
                  <React.Fragment key={`${viz.name.toLocaleLowerCase()}-citation-${index}`}>
                    {citation.fullCitation} ({<a href={citation.link}>link</a>})
                  </React.Fragment>
                ))}
              </List.Item>
              <List.Item>
                version: {viz.repo.version} (last updated {viz.repo.lastUpdate}),
                {<a href={viz.repo.link}> github link</a>}
              </List.Item>
              <List.Item>
                <Button basic={true} icon={true} labelPosition={'right'}>
                  <Link to={{ pathname: '/dataset', search: `?name=hpc/full&app=${viz.name.toLocaleLowerCase()}` }}>
                    {`launch ${viz.name}`}
                  </Link>
                  {/* Power Gap */}
                  <Icon name={'external alternate'} />
                </Button>
              </List.Item>
            </List>
          </>
        </Grid.Column>
      </Grid>
    );
  }

  protected renderExamples() {
    const panels = [
      {
        content: {
          content: (
            <List>
              <Divider />
              <List.Item>{this.renderExampleBlock('Example 1')}</List.Item>
              <List.Item>{this.renderExampleBlock('Example Dos')}</List.Item>
              <List.Item>{this.renderExampleBlock('Example San')}</List.Item>
              <List.Item>Example 2</List.Item>
            </List>
          ),
        },
        key: 'examples',
        title: 'examples',
      },
      {
        content: 'coming soon!',
        key: 'learn',
        title: 'learn',
      },
    ];

    return <Accordion panels={panels} />;
  }

  protected renderExampleBlock(title: string) {
    return (
      <Grid centered={true} columns={2}>
        <Grid.Column width={2}>
          <Icon name={'user secret'} size={'huge'} />
        </Grid.Column>
        <Grid.Column>
          <Message>
            <Message.Header>{title}</Message.Header>
            <Message.Content>
              <p>This is placeholder text - hooray!</p>
              <Button floated={'right'} icon={true}>
                <Icon name={'external alternate'} />
                launch example
              </Button>
            </Message.Content>
          </Message>
        </Grid.Column>
      </Grid>
    );
  }

  protected setupCurrentViz() {
    const params = new URLSearchParams(this.props.location ? this.props.location.search : '');
    // tslint:disable-next-line:no-backbone-get-set-outside-model
    const vizName = params.get('name');

    if (vizName === VizData.spring.name.toLocaleLowerCase()) {
      this.setState({
        currentViz: VizData.spring,
      });
    } else if (vizName === VizData.tfjsTsne.name.toLocaleLowerCase()) {
      this.setState({
        currentViz: VizData.tfjsTsne,
      });
    } else if (vizName === VizData.anatomogram.name.toLocaleLowerCase()) {
      this.setState({
        currentViz: VizData.anatomogram,
      });
    }
  }
}
