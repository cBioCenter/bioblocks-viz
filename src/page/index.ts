export * from './AppsPage';
export * from './DatasetPage';
export * from './VizOverviewPage';
