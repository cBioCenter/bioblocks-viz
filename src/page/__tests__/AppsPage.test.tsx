import { shallow } from 'enzyme';
import * as React from 'react';

import { AppsPage } from '~chell-viz~/page';

describe('AppsPage', () => {
  it('Should match existing snapshot.', () => {
    const wrapper = shallow(<AppsPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
