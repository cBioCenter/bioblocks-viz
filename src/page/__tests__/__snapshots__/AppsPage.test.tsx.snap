// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsPage Should match existing snapshot. 1`] = `
<List
  divided={true}
>
  <ListItem>
    <Grid
      centered={true}
      columns={3}
      padded={true}
      relaxed={true}
    >
      <GridColumn>
        <img
          alt="icon for SPRING"
          src="assets/icons/spring-thumbnail.png"
          style={
            Object {
              "height": "100px",
            }
          }
        />
      </GridColumn>
      <GridColumn
        textAlign="left"
      >
        <Header
          as="h2"
        >
          SPRING
        </Header>
        <List>
          <ListItem>
            caleb weinreb, samuel wolock, allon klein
          </ListItem>
          <ListItem>
            SPRING is a tool for uncovering high-dimensional structure in single-cell expression data.
          </ListItem>
          <ListItem>
            relevant data: scRNA-seq
          </ListItem>
        </List>
      </GridColumn>
      <GridColumn
        stretched={true}
      >
        <GridRow>
          <Button
            as="button"
            basic={true}
            role="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/overview",
                  "search": "?name=spring",
                }
              }
            >
              details
            </Link>
          </Button>
        </GridRow>
        <GridRow>
          <Button
            as="button"
            basic={true}
            role="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/dataset",
                  "search": "?app=spring",
                }
              }
            >
              launch
            </Link>
          </Button>
        </GridRow>
      </GridColumn>
    </Grid>
  </ListItem>
  <ListItem>
    <Grid
      centered={true}
      columns={3}
      padded={true}
      relaxed={true}
    >
      <GridColumn>
        <img
          alt="icon for tfjs-tsne"
          src="assets/icons/tfjs-tsne-thumbnail.png"
          style={
            Object {
              "height": "100px",
            }
          }
        />
      </GridColumn>
      <GridColumn
        textAlign="left"
      >
        <Header
          as="h2"
        >
          tfjs-tsne
        </Header>
        <List>
          <ListItem>
            original: Yannick Assogba
          </ListItem>
          <ListItem>
            Improved tSNE implementation that runs in the browser.
          </ListItem>
          <ListItem>
            relevant data: scRNA-seq
          </ListItem>
        </List>
      </GridColumn>
      <GridColumn
        stretched={true}
      >
        <GridRow>
          <Button
            as="button"
            basic={true}
            role="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/overview",
                  "search": "?name=tfjs-tsne",
                }
              }
            >
              details
            </Link>
          </Button>
        </GridRow>
        <GridRow>
          <Button
            as="button"
            basic={true}
            role="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/dataset",
                  "search": "?app=tfjs-tsne",
                }
              }
            >
              launch
            </Link>
          </Button>
        </GridRow>
      </GridColumn>
    </Grid>
  </ListItem>
  <ListItem>
    <Grid
      centered={true}
      columns={3}
      padded={true}
      relaxed={true}
    >
      <GridColumn>
        <img
          alt="icon for Anatomogram"
          src="assets/icons/anatomogram-thumbnail.png"
          style={
            Object {
              "height": "100px",
            }
          }
        />
      </GridColumn>
      <GridColumn
        textAlign="left"
      >
        <Header
          as="h2"
        >
          Anatomogram
        </Header>
        <List>
          <ListItem>
            original: Alfonso Mu√±oz-Pomer Fuentes
          </ListItem>
          <ListItem>
            Interactive anatomical diagram.
          </ListItem>
          <ListItem>
            relevant data: scRNA-seq
          </ListItem>
        </List>
      </GridColumn>
      <GridColumn
        stretched={true}
      >
        <GridRow>
          <Button
            as="button"
            basic={true}
            role="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/overview",
                  "search": "?name=anatomogram",
                }
              }
            >
              details
            </Link>
          </Button>
        </GridRow>
        <GridRow>
          <Button
            as="button"
            basic={true}
            role="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/dataset",
                  "search": "?app=anatomogram",
                }
              }
            >
              launch
            </Link>
          </Button>
        </GridRow>
      </GridColumn>
    </Grid>
  </ListItem>
</List>
`;
